src/main/include/controllers/DrivetrainTurningController.hpp

```diff
@@ -0,0 +1,194 @@
+     */
+    class Input {
+    public:
+        /// Left motor voltage.
+        static constexpr int kLeftVoltage = 0;
+
+        /// Right motor voltage.
+        static constexpr int kRightVoltage = 1;
+    };
+
+    /**
+     * Local outputs of the drivetrain system.
+     */
+    class LocalOutput {
+    public:
+        /// Heading in global coordinate frame.
+        static constexpr int kHeading = 0;
+
+        /// Left encoder position.
+        static constexpr int kLeftPosition = 1;
+
+        /// Right encoder position.
+        static constexpr int kRightPosition = 2;
+
+        /// Acceleration along X axis in chassis coordinate frame.
+        static constexpr int kAccelerationX = 3;
+
+        /// Acceleration along Y axis in chassis coordinate frame.
+        static constexpr int kAccelerationY = 4;
+    };
+
+    /**
+     * Constructs a drivetrain turning controller
+     */
+    DrivetrainTurningController();
+
+    /**
+     * Move constructor.
+     */
+    DrivetrainTurningController(DrivetrainTurningController&&) = default;
+
+    /**
+     * Move assignment operator.
+     */
+    DrivetrainTurningController& operator=(DrivetrainTurningController&&) =
+        default;
+
+    /**
+     * Sets the end goal of the controller profile.
+     *
+     * @param angle           Goal angle.
+     * @param angularVelocity Goal angular velocity.
+     */
+    void SetGoal(units::radian_t angle,
+                 units::radians_per_second_t angularVelocity);
+
+    /**
+     * Sets the references.
+     *
+     * @param angle  Angle of the carriage in radians.
+     * @param angularVelocity  Angular velocity of the carriage in radians per
+     *                         second.
+     */
+    void SetReferences(units::radian_t angle,
+                       units::radians_per_second_t angularVelocity);
+
+    /**
+     * Returns true if drivetrain controller has a new heading goal.
+     */
+    bool HaveHeadingGoal() const;
+
+    /**
+     *  Abort turn in place actions.
+     */
+    void AbortTurnInPlace();
+
+    /**
+     * Returns whether the drivetrain controller is at the goal heading.
+     */
+    bool AtHeading() const;
+
+    /**
+     * Returns the next output of the turn-in-place controller.
+     *
+     * @param x The current state x.
+     */
+    Eigen::Vector<double, 2> Calculate(
+        const Eigen::Vector<double, 3>& x) override;
+
+    /**
+     * Returns the drivetrain's plant.
+     */
+    static frc::LinearSystem<2, 2, 2> GetPlant();
+
+    /**
+     * The drivetrain system dynamics for turning in place.
+     */
+    static frc::LinearSystem<3, 2, 3> TurningDynamics();
+
+    /**
+     * Resets the controller.
+     */
+    void Reset(units::radian_t heading);
+
+private:
+    static constexpr auto kV = 1.382_V / 1_rad_per_s;
+    static constexpr auto kA = 0.08495_V / 1_rad_per_s_sq;
+
+    static constexpr auto kMaxV = 12_V / kV;
+    static constexpr auto kMaxA = 12_V / kA;
```

Comment:
Replace kV and kA with DrivetrainConstants::kAngularV and DrivetrainConstants::kAngularA.

Also, move kMaxV and kMaxA to DrivetrainConstants and rename them to kMaxAngularV and kMaxAngularA. After doing that, rename the preexisting kMaxV and kMaxA to kMaxLinearV and kMaxLinearA to avoid ambiguity.

Reply:
![Screenshot 2022-02-21 120306](https://user-images.githubusercontent.com/51429870/155019595-07b1f7c4-d75b-45d9-a673-5d51aefb4955.png)

Complains about the units being in meters.

Reply:
Here's the conversion from V/(rad/s) to V/(m/s). You could do that in reverse. https://github.com/wpilibsuite/allwpilib/blob/main/wpimath/src/main/native/include/frc/system/plant/LinearSystemId.h#L256-L290